### PR TITLE
fix: authorize nil value for object evaluation

### LINKED ISSUE #117: [BUG] When calling client.ObjectValueDetails with a value as nil the ResolutionDetail are not accessible.
### Observed behavior

I am using `nil` as a default value while calling `client.ObjectValueDetails` but when doing this everytime if the default value is used we receive no `ResolutionDetail` from the SDK.

After looking a bit at the SDK it seems that we add the `ResolutionDetail` only if the `value` is not nil.
https://github.com/open-feature/go-sdk/blob/fd678bd263eb532942c4777f4eed8e26a6e52694/pkg/openfeature/client.go#L642

It means that we cannot use `nil` as `value` in the SDK.

### Expected Behavior

I would like to be able to use `nil` as value for my flag.

### Steps to reproduce

SDK return a flag with a `nil` value.

### PR DESCRIPTION
**Pull request by thomaspoignant**
## This PR
Allow using a `nil` value when calling the `ObjectValueDetails` method.

### Related Issues
Fixes #117

### CODE REVIEW

File: pkg/openfeature/client_test.go [hunk 1]
```diff
@@ -866,3 +866,43 @@ func TestClientProviderLock(t *testing.T) {
...
+		Return(InterfaceResolutionDetail{
+			Value: value,
+			ProviderResolutionDetail: ProviderResolutionDetail{
+				Variant: variant,
+				Reason:  reason,
+			},
+		})
+
+	client := NewClient("test")
+	evDetails, _ := client.ObjectValueDetails(context.Background(), "foo", nil, EvaluationContext{})
```
Comment (skyerus) [inline pkg/openfeature/client_test.go:892]: Can we have a check on the error?

Comment (thomaspoignant) [reply pkg/openfeature/client_test.go]: Done

**Review by toddbaert — APPROVED** [review]
This makes sense to me.

I think once the errors are fixed it can be merged as long as all the tests pass.

**Comment by thomaspoignant** [comment]
I tried to fix everything from my phone, but it was not a good idea 🤦.
It should be fine now.

**Review by james-milligan — APPROVED** [review]

**Comment by toddbaert** [comment]
> I tried to fix everything from my phone, but it was not a good idea facepalm. It should be fine now.

Honestly I'm impressed you even attempted it!

**Review by toddbaert — APPROVED** [review]

**Pull request merged**